### ISSUE TITLE
Does ThinkTwice have Evaluation on Carla Leaderboard?

### ISSUE THREAD

**Issue by Naive-Bayes** [OpenDriveLab/ThinkTwice#2]
In paper, the method is evaluated on Town05 Long, however most of End-End method may report the eval result on Carla Leaderboard. So I want to know does ThinkTwice have evaluation on Carla Leaderboard?

**Comment by jiaxiaosong1002** [comment]
Thanks for your interests. Due to the complex environment dependency, we do not have a result currently. We will try it when we have bandwidth.

**Comment by Naive-Bayes** [comment]
Thanks for your replay. I find the time cost is so large no matter the collection or the evaluation on Carla, could you provide some useful method to accelerate this process?

**Comment by jiaxiaosong1002** [comment]
Yes, this is why we do not run 3 runs in Longest6.  As mentioned in [docs/DATA_PREP.md](https://github.com/OpenDriveLab/ThinkTwice/blob/main/docs/DATA_PREP.md), you could collect small images directly (remeber to change their cooresponding intrinsics).  Also, we observe that Lidar could make Carla run significantly slower. Thus, you could try to develop a camera-only model.

Note that compared to nuScenes (1000 scences * 20s), the normal size of Carla dataset (189K frames in TCP/ThinkTwice) is much larger. Thus, you may try to start with our ckeckpoint to initialize your backbone to reduce the difficulty of perception part.

Hope it could be helpful.

**Comment by exiawsh** [comment]
@jiaxiaosong1002 Hi, thanks for your excellent work. Because of the huge size of Carla dataset. It's challenging to follow your work. We want to do research on camera-only models, so would you please provide a baseline (189K frames) on a 4C model without lidar?

**Comment by jiaxiaosong1002** [comment]
Thanks for your interests. Currently, we do not have bandwidth/computational power to train/tune a new model.  Maybe you could consider load the weights of the checkpoint to a camera-only model and then finetune it, which could be easier.

**Comment by exiawsh** [comment]
> Thanks for your interests. Currently, we do not have bandwidth/computational power to train/tune a new model. Maybe you could consider load the weights of the checkpoint to a camera-only model and then finetune it, which could be easier.

Thank you for quickly reply. I will try it！

**Comment by exiawsh** [comment]
> Thanks for your interests. Currently, we do not have bandwidth/computational power to train/tune a new model. Maybe you could consider load the weights of the checkpoint to a camera-only model and then finetune it, which could be easier.

Hi, sorry to bother you. I have another question. How long does it take to generate 189k data?

**Comment by jiaxiaosong1002** [comment]
@exiawsh 
When we collected data, we use ~80 processes for about 1 week to collect 2 million frames in total. You could do the calculation. Note that the removement of Lidar might significantly increase speed. 

Another advice is that you could first only collect data from Town01, Town03, Town04, and Town06 for training as in Appendix B Table 1 of [TCP](https://arxiv.org/pdf/2206.08129.pdf), since the open-loop metric on validation set is not really related to the closed-loop evaluation results. As a result, you could approximately half the needed data.

For your information, when *current_throttle_brake_offset* < 0.1 and *current_longitudinal_offset* < 0.2 in training set, we consider the model fits the training set enough. Again, it is still not highly related to the closed-loop evaluation results.

**Comment by exiawsh** [comment]
> @exiawsh When we collected data, we use ~80 processes for about 1 week to collect 2 million frames in total. You could do the calculation. Note that the removement of Lidar might significantly increase speed.
> 
> Another advice is that you could first only collect data from Town01, Town03, Town04, and Town06 for training as in Appendix B Table 1 of [TCP](https://arxiv.org/pdf/2206.08129.pdf), since the open-loop metric on validation set is not really related to the closed-loop evaluation results. As a result, you could approximately half the needed data.
> 
> For your information, when _current_throttle_brake_offset_ < 0.1 and _current_longitudinal_offset_ < 0.2 in training set, we consider the model fits the training set enough. Again, it is still not highly related to the closed-loop evaluation results.

Thanks for your patient answer! I will try it!

**Comment by exiawsh** [comment]
> @exiawsh When we collected data, we use ~80 processes for about 1 week to collect 2 million frames in total. You could do the calculation. Note that the removement of Lidar might significantly increase speed.
> 
> Another advice is that you could first only collect data from Town01, Town03, Town04, and Town06 for training as in Appendix B Table 1 of [TCP](https://arxiv.org/pdf/2206.08129.pdf), since the open-loop metric on validation set is not really related to the closed-loop evaluation results. As a result, you could approximately half the needed data.
> 
> For your information, when _current_throttle_brake_offset_ < 0.1 and _current_longitudinal_offset_ < 0.2 in training set, we consider the model fits the training set enough. Again, it is still not highly related to the closed-loop evaluation results.

Hi, would you please share the scripts to generate the date using multi-processing? Thanks a lot!

**Comment by jiaxiaosong1002** [comment]
@exiawsh 
We collect our data in a cluster with slurm approximately one year ago.  I give the picture of the scripts as following:

![image](https://github.com/OpenDriveLab/ThinkTwice/assets/81464843/845a1025-6605-430a-a881-5b5d21a2f396)

![image](https://github.com/OpenDriveLab/ThinkTwice/assets/81464843/747b78f3-f88b-4212-bede-ff87ca0f015b)


![image](https://github.com/OpenDriveLab/ThinkTwice/assets/81464843/235ed8d2-e9b1-469c-a8a4-a13dd28e6321)

You could write your own script with some minor modification.
